src/main/duke/io/Parser.java

```diff
@@ -0,0 +1,121 @@
+package main.duke.io;
+
+import main.duke.commands.*;
+import main.duke.DukeException;
+
+import java.util.Arrays;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+public class Parser {
+
+    /**
+     * @param userInput the whole console command line from the user
+     * @return Command to be executed
+     * @throws DukeException when invalid input detected
+     */
+    public Command parse(String userInput) throws DukeException {
+        String[] inputArray = userInput.split(" ");
+        String userCommand = inputArray[0];
+        Command newCommand;
+
+        switch (userCommand) {
```

Comment:
there should not be an indentation for case clauses to follow the coding standards